### PR TITLE
Refactor authorization

### PR DESCRIPTION
**Pull request by Frawless**
Signed-off-by: Jakub Stejskal <xstejs24@gmail.com>

### Type of change

_Select the type of your PR_

- Bugfix
- Enhancement / new feature
- Refactoring
- Documentation

### Description

_Please describe your pull request_

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: systemtest/src/main/java/io/strimzi/systemtest/keycloak/KeycloakInstance.java [hunk 1]
```diff
@@ -154,6 +157,14 @@ public int getJwksRefreshSeconds() {
         return jwksRefreshSeconds;
     }
 
+    public static String getKeycloakSecretName() {
+        return KEYCLOAK_SECRET_NAME;
+    }
+
+    public static String getKeycloakSecretCert() {
+        return KEYCLOAK_SECRET_CERT;
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/keycloak/KeycloakInstance.java]: It is not a good practice, to static getters methods. You can easily change the access modifier of that instance variable from private to public.

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBasicClientResource.java [hunk 2]
```diff
@@ -49,40 +49,41 @@ public DoneableJob producerStrimzi() {
...
-                                        .withName("PRODUCER_ACKS")
-                                        .withValue("all")
-                                    .endEnv()
-                                    .addNewEnv()
-                                        .withName("ADDITIONAL_CONFIG")
-                                        .withValue(additionalConfig)
-                                    .endEnv()
+                                .withName(producerName)
+                                .withImagePullPolicy("Always")
+                                .withImage("frawless/hello-world-producer:latest")
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBasicClientResource.java]: This image will be changed right? Are you gonna to create a PR for that?

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBasicClientResource.java [hunk 3]
```diff
@@ -49,40 +49,41 @@ public DoneableJob producerStrimzi() {
...
+                                    .withName("TOPIC")
+                                    .withValue(topicName)
+                                .endEnv()
+                                .addNewEnv()
+                                    .withName("DELAY_MS")
+                                    .withValue(String.valueOf(delayMs))
+                                .endEnv()
+                                .addNewEnv()
+                                    .withName("LOG_LEVEL")
+                                    .withValue("INFO")
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBasicClientResource.java]: Why do not have we here DEBUG mode?

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBasicClientResource.java [hunk 4]
```diff
@@ -111,35 +112,36 @@ public DoneableJob consumerStrimzi() {
...
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBasicClientResource.java]: same here

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBridgeClientsResource.java [hunk 5]
```diff
@@ -23,7 +23,7 @@
 
     public KafkaBridgeClientsResource(String producerName, String consumerName, String bootstrapServer, String topicName,
                                       int messageCount, String additionalConfig, String consumerGroup, int port, int sendInterval, int pollInterval) {
-        super(producerName, consumerName, bootstrapServer, topicName, messageCount, additionalConfig, consumerGroup);
+        super(producerName, consumerName, bootstrapServer, topicName, messageCount, additionalConfig, consumerGroup, sendInterval);
         this.port = port;
         this.sendInterval = sendInterval;
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaBridgeClientsResource.java]: You should removed this instance variable `sendInterval` if you extends the  `KafkaClients` constructor. It is redundant here.

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java [hunk 6]
```diff
@@ -60,6 +60,12 @@ public static void waitForClientSuccess(String jobName, String namespace, int me
             () -> kubeClient().getClient().batch().jobs().inNamespace(namespace).withName(jobName).get().getStatus().getSucceeded().equals(1));
     }
 
+    public static void waitForClientFailure(String jobName, String namespace, long timeout) {
+        LOGGER.info("Waiting for producer/consumer:{} will be in error state", jobName);
+        TestUtils.waitFor("job finished", Constants.GLOBAL_POLL_INTERVAL, timeout,
+            () -> kubeClient().getClient().batch().jobs().inNamespace(namespace).withName(jobName).get().getStatus().getSucceeded().equals(1));
+    }
+
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:68]: Maybe we should take into consideration put these types of jobs to the JobUtils.class which you created?

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAbstractST.java [hunk 7]
```diff
@@ -70,7 +66,7 @@ void setup() throws Exception {
 
         LOGGER.info("Deploying keycloak...");
 
-        KeycloakUtils.deployKeycloak(NAMESPACE);
+//        KeycloakUtils.deployKeycloak(NAMESPACE);
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAbstractST.java]: So, are you not deploying the keycloak ? :D

File: systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAuthorizationST.java [hunk 8]
```diff
@@ -276,30 +275,28 @@ void setUp()  {
...
+                keycloakInstance.getOauthTokenEndpointUri());
+
+        teamBOauthClientJob = new KafkaOauthClientsResource(
+                TEAM_B_PRODUCER_NAME,
+                TEAM_B_CONSUMER_NAME,
+                KafkaResources.tlsBootstrapAddress(CLUSTER_NAME),
+                TOPIC_A,
+                MESSAGE_COUNT,
+                "",
+                "a-consumer_group",
```
Comment (see-quick) [inline systemtest/src/test/java/io/strimzi/systemtest/security/oauth/OauthAuthorizationST.java]: here should be `x-consumer_group` or?

**Review by see-quick — APPROVED** [review]
Few commends

**Review by see-quick — APPROVED** [review]

File: systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaOauthClientsResource.java [hunk 9]
```diff
@@ -67,8 +84,8 @@ public DoneableJob producerStrimziOauthPlain() {
                                 .withName("OAUTH_CRT")
                                 .editOrNewValueFrom()
                                     .withNewSecretKeyRef()
-                                        .withName("x509-https-secret")
-                                        .withKey("tls.crt")
+                                        .withName(KeycloakInstance.getKeycloakSecretName())
```
Comment (see-quick) [inline systemtest/src/main/java/io/strimzi/systemtest/resources/crd/kafkaclients/KafkaOauthClientsResource.java]: here just reference with the public static variable

File: systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java (hunk 6 above)
Comment (Frawless) [reply systemtest/src/main/java/io/strimzi/systemtest/utils/ClientUtils.java:68]: Next PR

**Pull request merged**